mne/viz/evoked.py

```diff
@@ -1566,120 +1475,149 @@ def _check_loc_legal(loc, what='your choice', default=1):
+        ranks = {val: ix for ix, val in enumerate(sorted(set(color_vals)))}
+        for key, orig_int in colors.items():
+            colors[key] = ranks[orig_int]
+        # if no cmap, convert color ints to real colors
+        if cmap is None:
+            color_list = _get_color_list()
+            for cond, color_int in colors.items():
+                colors[cond] = color_list[color_int]
+    # instantiate cmap
+    cmap_label = ''
+    if isinstance(cmap, (list, tuple, np.ndarray)) and len(cmap) == 2:
+        cmap_label, cmap = cmap
+    if isinstance(cmap, str):
+        cmap = get_cmap(cmap, lut=lut)
+    elif isinstance(cmap, Colormap) and all_int:
+        cmap = cmap._resample(lut)
+    # LINESTYLES
+    # make linestyles a list if it's not defined
+    if linestyles is None:
+        linestyles = [None] * len(conditions)  # will get assigned default
+    # convert linestyle list to dict
+    if isinstance(linestyles, (list, tuple, np.ndarray)):
+        if len(conditions) > len(linestyles):
+            raise ValueError('Trying to plot {} conditions, but there are '
+                             'only {} linestyles. Please specify linestyles '
+                             'manually.'
+                             .format(len(conditions), len(linestyles)))
+        linestyles = dict(zip(conditions, linestyles))
+    # should be a dict by now...
     if not isinstance(linestyles, dict):
-        linestyles = {condition: linestyle for condition, linestyle in
-                      zip(conditions, ['-'] * len(conditions))}
-    else:
-        linestyles = _aux_setup_styles(conditions, linestyles,
-                                       "linestyle", "-")
-
-    # finally, put it all together
-    styles = dict() if styles is None else deepcopy(styles)
-
-    for condition, color, linestyle in zip(conditions, colors, linestyles):
-        styles[condition] = styles.get(condition, dict())
-        styles[condition]['c'] = styles[condition].get('c', colors[condition])
-        styles[condition]['linestyle'] = styles[condition].get(
-            'linestyle', linestyles[condition])
-
-    return styles, the_colors, color_conds, color_order, colors_are_float
+        raise TypeError('"linestyles" must be a dict, list, or None; got {}.'
+                        .format(type(linestyles).__name__))
+    # validate linestyle dict keys
+    # (skip validating linestyle dict values... too hard to test)
+    if not set(linestyles).issubset(tags.union(conditions)):
+        raise ValueError('If "linestyles" is a dict its keys ({}) must '
+                         'match the keys/conditions in "evokeds" ({}).'
+                         .format(list(linestyles), conditions))
+    # transfer colors/linestyles dicts into styles dict; fall back on defaults
+    defaults = dict(color='gray', linestyle='solid')
+    current_styles = dict(color=colors, linestyle=linestyles)
+    for cond, inner_dict in styles.items():               # for each condition
```

Comment:
Oh god what is this :D

Reply:
I know, this is a beast, but I couldn't think of a better way.  Here's the goal: for each condition, go through the style types (this is just `('color', 'linestyle')`) and see if it was already given in the user-provided `styles` dict. To do that, we first look up the "full name" (e.g., `auditory/left`). If that fails, check each of the "tags" `('auditory', 'left')` separately. If that fails, assign a default value (`gray` or `solid`).

If you can think of a cleaner way to do this, I'm all ears :ear: :ear: :ear:

Reply:
For example, couldn't this be refactored with a function:

```
for style_name, curr in ...:
    inner_dict[style_name] = _setup_pce_style(inner_dict, style_name)
```